File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/python/non-vul/code_before31.py
# Source: Row 16 in ./dataset/CVEfixes/Analysis/results/Python/df_python_cwe_863.xlsx

def resolve_orders(root: models.User, info, **kwargs):
        from ..order.types import OrderCountableConnection

        def _resolve_orders(orders):
            requester = get_user_or_app_from_context(info.context)
            if not requester.has_perm(OrderPermissions.MANAGE_ORDERS):
                # allow fetch requestor orders (except drafts)
                if root == info.context.user:
                    orders = list(
                        filter(lambda order: order.status != OrderStatus.DRAFT, orders)
                    )
                else:
                    raise PermissionDenied()

            return create_connection_slice(
                orders, info, kwargs, OrderCountableConnection
            )

        return OrdersByUserLoader(info.context).load(root.id).then(_resolve_orders)

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/non-vul/code_before31.java
// Source: Row 474 in ./dataset/CVEfixes/Analysis/results/Java/df_java_cwe_all.xlsx

protected boolean isProbablePrime(BigInteger x, int iterations)
{
    /*
        * Primes class for FIPS 186-4 C.3 primality checking
        */
    return !Primes.hasAnySmallFactors(x) && Primes.isMRProbablePrime(x, param.getRandom(), iterations);
}